runtime/onert/core/src/backend/builtin/train/kernel/PermuteLayer.cc

```diff
@@ -31,16 +31,37 @@ namespace kernel
 
 PermuteLayer::PermuteLayer(const std::vector<ITensor *> &src_tensors,
                            const std::vector<ITensor *> &dst_tensors,
+                           const std::vector<ITensor *> &input_deriv_tensors,
+                           const std::vector<ITensor *> &output_deriv_tensors,
                            const std::shared_ptr<ExternalContext> &external_context)
-  : builtin::kernel::PermuteLayer{src_tensors, dst_tensors, external_context}
+  : builtin::kernel::PermuteLayer{src_tensors, dst_tensors, external_context},
+    _input_deriv_tensors{input_deriv_tensors}, _output_deriv_tensors{output_deriv_tensors}
 {
+  assert(input_deriv_tensors.size() == output_deriv_tensors.size());
+  assert(input_deriv_tensors.size() == src_tensors.size());
+}
+
+void PermuteLayer::optimize()
```

Comment:
Is this optimize() function called from train optimizer?

Reply:
No, this is for `PermuteLayer` only.